tests/validations/test_saved_query.py

```diff
@@ -87,6 +102,34 @@ def test_invalid_where_in_saved_query(  # noqa: D
     )
 
 
+def test_where_filter_validations_invalid_granularity(  # noqa: D
```

Comment:
based on the [similar test in the metrics validations tests](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/98b5af50ecf4c8d9730b3b24476cbec63e0aaad3/tests/validations/test_metrics.py#L460).